internal/driver/driver.go

```diff
@@ -88,27 +96,56 @@ func (s *Driver) HandleReadCommands(deviceName string, protocols map[string]mode
 
 			// check device is already initialized
 			if _, ok := s.generic[deviceLocation]; ok {
-				s.lc.Debugf(fmt.Sprintf("Driver.HandleReadCommands(): Device %v is already initialized with baud - %v, maxbytes - %v, timeout - %v", s.generic[deviceLocation], baudRate, key_maxbytes_value, key_timeout_value))
+				s.lc.Debugf("Driver.HandleReadCommands(): Device %v is already initialized with baud - %v, maxbytes - %v, timeout - %v", s.generic[deviceLocation], baudRate, key_maxbytes_value, key_timeout_value)
 			} else {
 				// initialize device for the first time
-				s.generic[deviceLocation] = NewUartGeneric(deviceLocation, baudRate, key_timeout_value)
+				s.generic[deviceLocation] = NewUartGeneric(deviceLocation, baudRate, key_timeout_value, s.lc)
 				s.generic[deviceLocation].rxbuf = nil
 
-				s.lc.Debugf(fmt.Sprintf("Driver.HandleReadCommands(): Device %v initialized for the first time with baud - %v, maxbytes - %v, timeout - %v", s.generic[deviceLocation], baudRate, key_maxbytes_value, key_timeout_value))
+				s.lc.Debugf("Driver.HandleReadCommands(): Device %v initialized for the first time with baud - %v, maxbytes - %v, timeout - %v", s.generic[deviceLocation], baudRate, key_maxbytes_value, key_timeout_value)
 			}
 
-			if err := s.generic[deviceLocation].GenericUartRead(key_maxbytes_value); err != nil {
+			if err := s.generic[deviceLocation].GenericUartRead(key_maxbytes_value, s.lc); err != nil {
 				return nil, fmt.Errorf("Driver.HandleReadCommands(): Reading UART failed: %v", err)
 			}
 
 			rxbuf := hex.EncodeToString(s.generic[deviceLocation].rxbuf)
-			s.lc.Debugf(fmt.Sprintf("Driver.HandleReadCommands(): Received Data =  %s", rxbuf))
+			s.lc.Infof("Driver.HandleReadCommands(): Received Data = %s", rxbuf)
```

Comment:
Avoid spamming the logs in production

```suggestion
			s.lc.Debugf("Driver.HandleReadCommands(): Received Data = %s", rxbuf)
```

Reply:
Addressed.